src/plugins/kernel_install/utils.sh

```diff
@@ -550,32 +550,85 @@ function do_uninstall()
   fi
 }
 
+# Element existence check
+# Checks if element belongs to given array
+#
+# @target_element element being checked
+# @array target array
+# returns 1 if element is in array and 0 otherwise
+function is_in_array()
+{
+  local target_element="$1"
+  local array=("${!2}")
```

Comment:
When dealing with indexed/associative arrays in Bash that you need a function to consult/populate/manipulate them, prefer the use of the form `local -n _reference` to create references to arrays declared outside the scope of these functions. In practical terms, you can make this line
```bash
local -n _array="$2"
```
and call the function below like `is_in_array "$kernel" 'kw_managed_installed_kernels'`.